internal/resource/user.go

```diff
@@ -0,0 +1,146 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+
+	"terraform-provider-plural/internal/common"
+	"terraform-provider-plural/internal/model"
+
+	"terraform-provider-plural/internal/client"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+)
+
+var _ resource.Resource = &UserResource{}
+var _ resource.ResourceWithImportState = &UserResource{}
+
+func NewUserResource() resource.Resource {
+	return &UserResource{}
+}
+
+// UserResource defines the User resource implementation.
+type UserResource struct {
+	client *client.Client
+}
+
+func (r *UserResource) Metadata(
+	_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse,
+) {
+	resp.TypeName = req.ProviderTypeName + "_user"
+}
+
+func (r *UserResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "user resource",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "Internal identifier of this user.",
+				MarkdownDescription: "Internal identifier of this user.",
+				Computed:            true,
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"name": schema.StringAttribute{
+				Description:         "Name of this user.",
+				MarkdownDescription: "Name of this user.",
+				Required:            true,
+			},
+			"email": schema.StringAttribute{
+				MarkdownDescription: "Email address of this user.",
+				Required:            true,
+			},
+		},
+	}
+}
+
+func (r *UserResource) Configure(
+	_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse,
+) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*common.ProviderData)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected User Resource Configure Type",
+			fmt.Sprintf(
+				"Expected *common.ProviderData, got: %T. Please report this issue to the provider developers.",
+				req.ProviderData,
+			),
+		)
+
+		return
+	}
+
+	r.client = data.Client
+}
+
+func (r *UserResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	data := new(model.User)
+	resp.Diagnostics.Append(req.Plan.Get(ctx, data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	response, err := r.client.UpdateUser(ctx, nil, data.Attributes())
```

Comment:
this definitely won't work, what happens when you don't specify the id, is it updates the current user.  I'll make sure the CreateUser resource is present for this at least.

Reply:
Oh I see the mutation was never actually defined, adding that to one of my prs, then we can use it here.